misc/git-hook/necessary/pre-commit

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Usage: add this file to your project's .git/hooks directory.
+# Now, when you change some files in repository and try to commit these
+# changes, git will run this script right before commit. 
+
+GITDIR=`git rev-parse --git-common-dir`
+if [ "$(expr substr $(uname -s) 1 10)" == "MINGW64_NT" ]; then
+    GITDIR=`cygpath --unix $GITDIR`
+fi
+
+export GITDIR
+
+if [ -f "./functions.bash" ]
+then
+    . ./functions.bash
+else
+    . $GITDIR/hooks/functions.bash
+fi
+
+echo ">>>>>>>> modifying begin <<<<<<<<"
+
+files=`find $GITDIR/hooks/precommit/modify/* -type f -maxdepth 0 -mindepth 0`
```

Comment:
This line gives a 'no such file or directory' issue when the optional tools are not installed.
![image](https://user-images.githubusercontent.com/3918517/165140479-80357081-4735-45ed-8c48-1a3192ca98f7.png)